zentral/contrib/osquery/serializers.py

```diff
@@ -166,17 +167,15 @@ def validate(self, data):
         if data.get("compliance_check_enabled"):
             sql = data.get("sql")
             if sql is not None and 'ztl_status' not in sql:
-                raise serializers.ValidationError('{compliance_check_enabled: true} '
-                                                  'only if query contains ztl_status')
+                raise serializers.ValidationError({'sql': 'ztl_status not in sql'})
```

Comment:
The error would be better in the context of the `compliance_check_enabled` field